lib/MusicBrainz/Server/Form/Utils.pm

```diff
@@ -205,6 +206,10 @@ sub validate_username {
 
     if (defined $username) {
         unless (defined $previous_username && $editor_model->are_names_equivalent($previous_username, $username)) {
+            my $sanitized_name = sanitize($username);
+            if ($username ne $sanitized_name) {
+                $self->add_error(l('This username contains invalid characters. (Check for leading, trailing, or consecutive spaces.)'));
```

Comment:
The only improvement I could think of would be to give the user the sanitized username back. Something like:

`This username contains invalid characters. The closest valid form of the username is {sanitized_username} - feel free to use that`

But dunno, maybe it's weird.

Reply:
Only issue I see is that the sanitized name might be taken.  I did notice that the form trims the username anyway, so as far as whitespace is concerned, only consecutive spaces should need to be checked.

Reply:
You *could* check for that and give them a separate message if it is taken, FWIW. We already have a check after all :)

Reply:
Right, but I'm not sure if the situation is common enough to warrant having a custom message for it.  They may not even want to use the sanitized name we suggest. :)

Reply:
Implementing a custom message would be over-engineered indeed. The new check is mainly to prevent entering control characters, I don’t expect genuine users to see it; See on IRC.